spec/integration/user_post_spec.rb

```diff
@@ -0,0 +1,95 @@
+
+  it 'display users first 3 posts' do
+    Post.create(id: 103, author: @user1, title: 'Post title 1 by User 1',
+                text: 'Post 1 by User 1', commentscounter: 0, likescounter: 0)
+    Post.create(author: @user1, title: 'Post title 2 by User 1',
+                text: 'Post 2 by User 1', commentscounter: 0, likescounter: 0)
+    Post.create(author: @user1, title: 'Post title 3 by User 1',
+                text: 'Post 3 by User 1', commentscounter: 0, likescounter: 0)
+    visit user_path(@user1)
+    sleep(5)
+    expect(page).to have_content('Post 1 by User 1')
+    expect(page).to have_content('Post 2 by User 1')
+    expect(page).to have_content('Post 3 by User 1')
+    expect(page).to have_css('a.see-more', count: 1)
+  end
+
+  it 'redirects to see all user posts' do
+    visit user_path(@user1)
+    click_link('See all posts')
+    sleep(5)
+    expect(current_path).to eq(user_posts_path(user_id: @user1.id))
+  end
+
+  it 'display user post index page' do
+    post1 = Post.create(id: 105, author: @user1, title: 'Post title 1 by User 1',
+                        text: 'Post 1 by User 1', commentscounter: 0, likescounter: 0)
+    post2 = Post.create(id: 106, author: @user1, title: 'Post title 2 by User 1',
+                        text: 'Post 2 by User 1', commentscounter: 0, likescounter: 0)
+    Comment.create(post: post1, user: @user1, text: 'User 1 sample comment 1')
+    Comment.create(post: post1, user: @user2, text: 'User 2 sample comment 1')
+    Comment.create(post: post2, user: @user1, text: 'User 1 sample comment 1')
+    Comment.create(post: post2, user: @user2, text: 'User 2 sample comment 1')
+    visit user_posts_path(user_id: @user1.id)
+    sleep(5)
+    expect(page).to have_content('Username 1')
+    find("img[src='test1.jpg']")
+    expect(page).to have_content('User 1 sample comment 1')
+    expect(page).to have_content('Comments: 2')
+    expect(page).to have_content('Likes: 0')
+    expect(page).to have_content('Pagination')
+  end
+
+  it 'redirect to the post show page' do
+    post1 = Post.create(id: 103, author: @user1, title: 'Post title 1 by User 1',
+                        text: 'Post 1 by User 1', commentscounter: 0, likescounter: 0)
+    Comment.create(post: post1, user: @user1, text: 'User 1 sample comment 1')
+
+    visit user_post_path(user_id: @user1.id, id: post1.id)
+    sleep(5)
+    expect(page).to have_content("Post ##{post1.id} by #{post1.author.name}")
+    expect(page).to have_content('Likes: 0')
+    expect(page).to have_content('Username 1: User 1 sample comment 1')
+  end
+
+  it 'redirect to user show page on clicking post' do
+    post1 = Post.create(id: 103, author: @user1, title: 'Post title 1 by User 1',
+                        text: 'Post 1 by User 1', commentscounter: 0, likescounter: 0)
+    visit user_posts_path(user_id: @user1.id)
+    click_link('Post 1 by User 1')
+    sleep(5)
+    expect(current_path).to eq(user_post_path(user_id: @user1.id, id: post1.id))
+  end
```

Comment:
- [Optional] I think you're missing featured tests for User post show page. It seems the previous reviewer didn't notice this but it is highly recommended to add them. This will make sure our app is functioning perfectly in the future :+1: 
![Screenshot from 2023-11-10 08-08-01](https://github.com/devalibello/blog-app/assets/63087491/5fbf8ee7-42d6-4eb2-85e2-5c816bfdfd09)